#VRML_SIM R2021b utf8
WorldInfo {
  title "ArduPilot Webots Simulator"
  gravity 9.80665
  physics "my_physics01"
  basicTimeStep 1
  FPS 15
  optimalThreadCount 4
  coordinateSystem "NUE"
  randomSeed 52
}
Viewpoint {
  orientation -0.997409773021728 0.07014477373751193 0.015920282599834967 0.4201670646038245
  position 0.10290187556190972 4.567903261359225 6.919647967286325
  follow "quad_plus_sitl"
  followType "Mounted Shot"
}
DogHouse {
  translation 34.82 0.76 -24.56
  name "dog house(1)"
}
DogHouse {
  translation 161.819 0.75 -152.174
  name "dog house(2)"
}
DogHouse {
  translation 185.42 0.77 48.97
  name "dog house(5)"
}
Background {
  skyColor [
    0.15 0.5 1
  ]
}
Solid {
  translation 36.93 0.77 -37.93
  children [
    HouseWithGarage {
    }
  ]
}
Solid {
  translation 192.76999999999998 0 64.98
  rotation 0 1 0 -1.5707963071795863
  children [
    HouseWithGarage {
    }
  ]
  name "solid(1)"
}
DEF DEF_VEHICLE Robot {
  translation -0.02764715934988559 0.6671153317430623 0.005104790304628726
  rotation 0.8479346967484522 -0.0006037865658242816 0.5301003541706926 0.0017289323074569565
  children [
    Emitter {
      name "emitter_plugin"
      description "commuicates with physics plugin"
    }
    Shape {
      appearance Appearance {
        material Material {
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
    Camera {
      translation 0 0.12 0
      rotation 0.12942795977353752 0.9831056944488316 0.12942795977353752 -1.5878343071795866
      name "camera1"
      width 128
      height 128
    }
    Compass {
      name "compass1"
    }
    GPS {
      name "gps1"
      type "laser"
    }
    Accelerometer {
      name "accelerometer1"
    }
    Gyro {
      name "gyro1"
    }
    InertialUnit {
      name "inertial_unit"
    }
    Transform {
      translation -0.09999999999999999 0 0
      rotation -0.5773502691896258 0.5773502691896258 0.5773502691896258 2.094395
      children [
        Solid {
          translation 0 0.2 0
          rotation 1 0 0 1.5707959999999999
          children [
            Propeller {
              shaftAxis 0 -1 0
              thrustConstants -1.01 0
              device RotationalMotor {
                name "motor3"
                controlPID 10.001 0 0
                maxVelocity 5000
              }
              fastHelix Solid {
                children [
                  Shape {
                    appearance Appearance {
                      material Material {
                        diffuseColor 1 0 0.1
                      }
                    }
                    geometry Cylinder {
                      height 0.002
                      radius 0.1
                    }
                  }
                ]
              }
              slowHelix Solid {
                rotation 0 1 0 1.1667874781290464
                children [
                  Shape {
                    appearance Appearance {
                      material Material {
                        diffuseColor 0.960784 0.47451 0
                      }
                    }
                    geometry Cylinder {
                      height 0.002
                      radius 0.1
                    }
                  }
                ]
              }
            }
          ]
          physics Physics {
            mass 0.05
          }
        }
        Shape {
          appearance Appearance {
            material Material {
            }
          }
          geometry DEF DEF_ARM Cylinder {
            height 0.1
            radius 0.01
          }
        }
      ]
    }
    Transform {
      translation 0 0 0.09999999999999999
      rotation 0 0.7071067811865476 0.7071067811865476 -3.1415923071795864
      children [
        Solid {
          translation 0 0.2 0
          rotation 1 0 0 1.5707959999999999
          children [
            Propeller {
              shaftAxis 0 1 0
              thrustConstants 1.01 0
              device RotationalMotor {
                name "motor2"
                controlPID 10.001 0 0
                maxVelocity 5000
              }
              fastHelix Solid {
                children [
                  Shape {
                    appearance Appearance {
                      material Material {
                        diffuseColor 1 0 0.1
                      }
                    }
                    geometry Cylinder {
                      height 0.002
                      radius 0.1
                    }
                  }
                ]
              }
              slowHelix Solid {
                rotation 0 1 0 1.1667874781290464
                children [
                  Shape {
                    appearance Appearance {
                      material Material {
                        diffuseColor 0.960784 0.47451 0
                      }
                    }
                    geometry Cylinder {
                      height 0.002
                      radius 0.1
                    }
                  }
                ]
              }
            }
          ]
          name "solid(2)"
          physics Physics {
            mass 0.05
          }
        }
        Shape {
          appearance Appearance {
            material Material {
            }
          }
          geometry USE DEF_ARM
        }
      ]
    }
    Transform {
      translation 0.09999999999999999 0 0
      rotation 0.5773502691896258 0.5773502691896258 0.5773502691896258 -2.094395307179586
      children [
        Solid {
          translation 0 0.2 0
          rotation 1 0 0 1.5707959999999999
          children [
            Propeller {
              shaftAxis 0 -1 0
              thrustConstants -1.01 0
              device RotationalMotor {
                name "motor1"
                controlPID 10.001 0 0
                maxVelocity 5000
              }
              fastHelix Solid {
                children [
                  Shape {
                    appearance Appearance {
                      material Material {
                        diffuseColor 1 0 0.1
                      }
                    }
                    geometry Cylinder {
                      height 0.002
                      radius 0.1
                    }
                  }
                ]
              }
              slowHelix Solid {
                rotation 0 1 0 1.1667874781290464
                children [
                  Shape {
                    appearance Appearance {
                      material Material {
                        diffuseColor 0.960784 0.47451 0
                      }
                    }
                    geometry Cylinder {
                      height 0.002
                      radius 0.1
                    }
                  }
                ]
              }
            }
          ]
          name "solid(1)"
          physics Physics {
            mass 0.05
          }
        }
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 1 0.09999999999999999 0
            }
          }
          geometry USE DEF_ARM
        }
      ]
    }
    Transform {
      translation 0 0 -0.09999999999999999
      rotation 1 0 0 -1.5707963071795863
      children [
        Solid {
          translation 0 0.2 0
          rotation 1 0 0 1.5707959999999999
          children [
            Propeller {
              shaftAxis 0 1 0
              thrustConstants 1.01 0
              device RotationalMotor {
                name "motor4"
                controlPID 10.001 0 0
                maxVelocity 5000
              }
              fastHelix Solid {
                children [
                  Shape {
                    appearance Appearance {
                      material Material {
                        diffuseColor 1 0 0.1
                      }
                    }
                    geometry Cylinder {
                      height 0.002
                      radius 0.1
                    }
                  }
                ]
              }
              slowHelix Solid {
                rotation 0 1 0 1.1667874781290464
                children [
                  Shape {
                    appearance Appearance {
                      material Material {
                        diffuseColor 0.960784 0.47451 0
                      }
                    }
                    geometry Cylinder {
                      height 0.002
                      radius 0.1
                    }
                  }
                ]
              }
            }
          ]
          name "solid(3)"
          physics Physics {
            mass 0.05
          }
        }
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0.7999999999999999 0.7999999999999999 0.7999999999999999
            }
          }
          geometry USE DEF_ARM
        }
      ]
    }
  ]
  name "quad_plus_sitl"
  boundingObject Box {
    size 0.1 0.1 0.1
  }
  physics Physics {
    density -1
    mass 1
    centerOfMass [
      0 0 0
    ]
  }
  rotationStep 0.261799
  controller "my_controller"
  controllerArgs [
    "-p"
    "5599"
    "-df"
    "0.01"
  ]
  customData "1"
  linearVelocity 3.3005427500547874e-05 5.661450261040858e-08 -1.7975378593927228e-05
  angularVelocity -0.0003603706832993974 6.421461067855578e-07 -0.0006606383403292716
}
DirectionalLight {
  direction 0 -1 0
}
UnevenTerrain {
  size 500 1 500
}
HouseWithGarage {
  translation 174.25 1.88 -157.5
  rotation 0 1 0 -1.5707963071795863
}
AdvertisingBoard {
  translation 84.03999999999999 2.35 -5.81
  rotation 0 1 0 -1.5707963071795863
  name "advertising board(1)"
  frontTexture [
    "https://ardupilot.org/application/files/6315/7552/1962/ArduPilot-Motto.png"
  ]
}
